integrations/helm/http/daemon/server.go

```diff
@@ -46,3 +56,27 @@ func ListenAndServe(listenAddr string, logger log.Logger, stopCh <-chan struct{}
 		logger.Log("info", "HTTP server stopped")
 	}
 }
+
+// NewHandler registers handlers on the given router.
+func NewHandler(s api.Server, r *mux.Router) http.Handler {
+	handle := APIServer{s}
+	r.Get(transport.SyncGit).HandlerFunc(handle.SyncGit)
+	return r
+}
+
+type APIServer struct {
+	server api.Server
+}
+
+// SyncGit starts a goroutine in the background to sync all git mirrors
+// and writes back a HTTP 200 status header and 'OK' body to inform it
+// has been started.
+// TODO(hidde): in the future we may want to give users the option to
+// request the status after it has been started. The Flux (daemon) API
+// archives this by working with jobs whos IDs can be tracked.
+func (s APIServer) SyncGit(w http.ResponseWriter, r *http.Request) {
+	go s.server.SyncMirrors()
```

Comment:
In general it's undesirable to start longish-running goroutines in response to user requests, without any backpressure or rate limiting (granted, there are precedents elsewhere in this code). A way to limit its effects here would be to succeed trivially if a sync is already in progress.